main/zoe/guide/contracts/nft-drop.md

```diff
@@ -0,0 +1,107 @@
+∫# NFT Drop
+
+<Zoe-Version/>
+
+##### [View the code on Github](https://github.com/Agoric/dapp-nft-drop/blob/main/contract/src/NFTDropContractBasic.js)
+##### [View all contracts on Github](https://github.com/Agoric/agoric-sdk/tree/master/packages/zoe/src/contracts)
+
+This contract provides users with an invitation that will mint a NFT in exchange for a fixed price. It demonstrates how one could use Zoe to facilitate the release, or "drop", of a set oF NFTs created for a particular project. The contract code results in the creation of an [invitation](../../../glossary/#invitation) which can be accessed via the public in order to purchase one of the NFTs in the collection.
+
+The purpose of the `NFTDropContractBasic.js` is to provide a point of reference for developers seeking to launch an NFT drop on the Agoric blockchain. While the implementation details are in complete control of the developer, some features that one may want to extend this contract with are:
+  * Fixing the total supply of NFTs available.
+  * Specifiying [the time](../../../repl/timerServices.md#timer-services):
+    * in which the NFT drop is open to the public.
+    * of an NFT drop deadline, after which users will no longer be able to mint and/or purchase.
+  * Embedded additional metadata relating the project.
+  * Whatever your heart desires!
+
+The code for the smart contract can be found below:
+```js
+/** @type {ContractStartFn} */
+const start = async (zcf) => {
+  const { pricePerNFT, nftName } = zcf.getTerms();
+
+  // Set up the mint
+  const mint = await zcf.makeZCFMint(nftName, AssetKind.SET);
+  const { brand: NFTBrand } = mint.getIssuerRecord();
+
+  let currentId = 1n;
+  const { zcfSeat: sellerSeat } = zcf.makeEmptySeatKit();
+
+  /** @type {OfferHandler} */
+  const buyNFTs = (buyerSeat) => {
+    // Mint the NFTs
+    const amount = AmountMath.make(NFTBrand, harden([currentId]));
+    mint.mintGains(harden({ NFTs: amount }), buyerSeat);
+    currentId += 1n;
+
+    // Take the money
+    sellerSeat.incrementBy(buyerSeat.decrementBy(harden({ Money: pricePerNFT })));
+
+    zcf.reallocate(buyerSeat, sellerSeat);
+    buyerSeat.exit();
+
+    return 'your offer was successful';
+  };
+
+  const publicFacet = Far('NFT Drop', {
+    makeInvitation: () => zcf.makeInvitation(buyNFTs, 'buyNFTs'),
+  });
+
+  return harden({ publicFacet });
+};
+
+```
+
+Taking a look inside of the contract code, you will notice that the minting process occurs **at the time of purchase**. This approach is in contrast to one in which a set of NFTs are minted prior to being made available to the public for purchase. 
+
+To accomplish this design, the contract first creates an internal `mint` variable. Note that the `mint` variable specifies the [`AssetKind`](../../../ertp/api/issuer.md#makeissuerkit-allegedname-assetkind-displayinfo) to be a `SET` which is necessary when creating NFTs with Zoe.
```

Comment:
`mint` is `const`, so calling it a variable seems odd.

And `AssetKind` is more of an ERTP level thing than Zoe. "with ERTP" is probably safe to just leave implicit.

```suggestion
To accomplish this design, the contract first creates an internal `mint` object. Note that the `mint` variable specifies the [`AssetKind`](../../../ertp/api/issuer.md#makeissuerkit-allegedname-assetkind-displayinfo) to be a `SET` which is necessary when creating NFTs.
```

Reply:
The `const` keyword is used to defined variables in JavaScript with the added behavior of variables defined using `const` can not be redeclared.

I understand it may seem odd, especially with the `var` keyword being the way in which variables were defined in JS for so long. The release of ES6 eliminated wide spread use of using `var` in exchange for `const` as well as `let`.

Reply:
> The `const` keyword is used to defined variables in JavaScript ...

Really? I don't recall any JS docs that speak of `const` declarations declaring variables.

See, for example [const in MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) that distinguishes constants from variables:
> Constants are block-scoped, much like variables declared using the let keyword. --

Reply:
Hmmm... yeah, i never really gave it much thought but this insinuates `const` is not used for declaring variables. Taking a look at the [Declarations](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#declarations) section of the MDN docs shows the following though:


> ### JavaScript has three kinds of variable declarations.
>
> **var**
>
>    Declares a variable, optionally initializing it to a value.
>
> **let**
>
>  Declares a block-scoped, local variable, optionally initializing it to a value.
>
> **const**
>
>    Declares a block-scoped, read-only named constant.

lol very odd right? i'm fine with changing the text, but again, I believe the majority of JS developers interpret `const` as a way to declare variables (at least the ones who have worked primarily in the last decade or so). I guess the question remains - what is the answer here?

Reply:
Initially i referenced info i got from glancing at w3schools on Google. 

https://www.w3schools.com/js/js_const.asp

I know Agoric has some representation on TC39. Maybe they could patch up the confusion for us 🤔🙂. 

For right now though i do feel as though the text won't be intepreted oddly by JS devs but again, happy to change it if we think it will cause confusion.